content/en/developers/authorization/oauth2_in_datadog.md

```diff
@@ -0,0 +1,107 @@
+---
+title: OAuth2 in Datadog
+kind: documentation
+description: Learn about how Datadog uses OAuth 2.0.
+further_reading:
+- link: "/developers/authorization/oauth2_endpoints"
+  tag: "Documentation"
+  text: "Learn how to use the OAuth 2.0 authorization endpoints"
+---
+
+{{< beta-callout btn_hidden="true" >}}
+  The Datadog Developer Platform is in beta. If you don't have access, contact apps@datadoghq.com.
+{{< /beta-callout >}} 
+
+## Overview
+
+This page provides a step-by-step overview on how to implement the OAuth protocol end-to-end on your application once your **confidential** client is created. 
+
+### Implement the OAuth protocol
+
+1. Create and configure your OAuth client in the [Developer Platform][16]. 
+
+2. After a user installs your integration, they can click on the **Connect Accounts** button which connects their account on the **Configure** tab in the integration tile. 
+
+   When a user clicks this button, they are directed to the `onboarding_url` that you provided as a part of the OAuth client creation process. This page should be the sign-in page for your platform.
+
+3. Once a user signs in, redirect them to the [OAuth2 Authorization endpoint][6] with the appropriate URL parameters, which includes the added `code_challenge` parameter generated by your application.
+
+   To learn how to derive the `code_challenge` parameter, see the [PKCE](#pkce) section. Your application must save `code_verifier` for the token request in Step 5.
+
+   In order to build this URL, use the `site` query parameter that is provided on the redirect to your `onboarding_url`. The format of the domain should match the following format. You can append the `authorize` path to this constructed URL: `https://{<custom_subdomain>,app,us3,us5}.{datadoghq,ddog-gov}.{com,eu}`.
```

Comment:
Did we decide that partners should always route to app.datadoghq.com (i.e. us1) so that they can use the data center dropdown to change data centers if needed? If so this part is optional / only relevant when navigating to the `onboarding_url` from datadog. I'm not sure we should call this out at all though if they can / should always route to us1.